netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class HandlePackageFileConflicts : Task
+    {
+        private Dictionary<string, int> packageRanks = null;
+
+        [Required]
+        public ITaskItem[] References { get; set; }
+
+        [Required]
+        public ITaskItem[] ReferenceCopyLocalPaths { get; set; }
+
+        /// <summary>
+        /// NuGet3 and later only.  In the case of a conflict with identical file version information a file from the most preferred package will be chosen.
+        /// </summary>
+        public string[] PreferredPackages { get; set; }
+
+        [Output]
+        public ITaskItem[] ReferencesWithoutConflicts { get; set; }
+
+        [Output]
+        public ITaskItem[] ReferenceCopyLocalPathsWithoutConflicts { get; set; }
+
+        public override bool Execute()
+        {
+            ReferencesWithoutConflicts = HandleConflicts(References);
+
+            ReferenceCopyLocalPathsWithoutConflicts = HandleConflicts(ReferenceCopyLocalPaths);
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private void EnsurePackageRanks()
+        {
+            if (packageRanks == null)
+            {
+                var numPrefferredPackages = PreferredPackages?.Length ?? 0;
+
+                // cache ranks for fast lookup
+                packageRanks = new Dictionary<string, int>(numPrefferredPackages, StringComparer.OrdinalIgnoreCase);
+
+                for (int i = 0; i < numPrefferredPackages; i++)
+                {
+                    var prefferedPackageId = PreferredPackages[i];
+
+                    if (!String.IsNullOrWhiteSpace(prefferedPackageId))
+                    {
+                        // handle duplicates
+                        if (!packageRanks.ContainsKey(prefferedPackageId))
```

Comment:
can't you just set via the indexer and not bother looking to see whether it's there or not